### PR TITLE
ci-operator/step-registry/ipi/deprovision/vsphere: Decompose into sub-steps

### PR DESCRIPTION
**Pull request by wking**
Builds on #7931; review that first.

So we can recycle the vanilla IPI deprovision step without needing to make changes to it (other than using the UPI image to get the vSphere CA certs).  This also makes the link between the resource-creating and resource-destroying steps more transparent.

Also store the hosted zone ID in a secret so we don't have to look it up again, which lets us simplify the secret loading a bit in the DNS deprovisioning.

Remove the ipi-deprovision-vsphere chain, since there's no reason folks would use it instead of the ipi-vsphere-post chain.

### CODE REVIEW

**Comment by wking** [comment]
> Builds on #7931; review that first.

Rebased onto master to drop the dependency on #7931, while that gets sorted out.

File: ci-operator/step-registry/ipi/deprovision/vsphere/dns/ipi-deprovision-vsphere-dns-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: ipi-deprovision-vsphere-dns
+  from: upi-installer
+  commands: ipi-deprovision-vsphere-dns-commands.sh
+  resources:
+    requests:
+      cpu: 1000m
+      memory: 100Mi
+  documentation: >-
+    Releases DNS records via `${SHARED_DIR}/dns-delete.json`.  The IP addresses were allocated with the ipi-conf-vsphere-dns step.
```
Comment (abhinavdahiya) [inline ci-operator/step-registry/ipi/deprovision/vsphere/dns/ipi-deprovision-vsphere-dns-ref.yaml]: mention the `${SHARED_DIR}/hosted-zone.tx`

**Comment by abhinavdahiya** [comment]
LGTM

/assign @patrickdillon

File: ci-operator/step-registry/ipi/deprovision/vsphere/dns/ipi-deprovision-vsphere-dns-ref.yaml (hunk 1 above)
Comment (wking) [reply ci-operator/step-registry/ipi/deprovision/vsphere/dns/ipi-deprovision-vsphere-dns-ref.yaml]: > mention the `${SHARED_DIR}/hosted-zone.tx`

Done with 71b216028 -> a466b8351.

**Comment by wking** [comment]
[vsphere rehearsal][1] failed waiting for the Kube API:

```
level=info msg="Waiting up to 20m0s for the Kubernetes API at https://api.ci-op-75qqryhn-6de38.origin-ci-int-aws.dev.rhcloud.com:6443..."
level=error msg="Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get https://api.ci-op-75qqryhn-6de38.origin-ci-int-aws.dev.rhcloud.com:6443/apis/config.openshift.io/v1/clusteroperators: dial tcp: lookup api.ci-op-75qqryhn-6de38.origin-ci-int-aws.dev.rhcloud.com on 10.142.0.51:53: server misbehaving"
level=info msg="Pulling debug logs from the bootstrap machine"
level=info msg="Bootstrap gather logs captured here \"/tmp/installer/log-bundle-20200326001715.tar.gz\""
level=fatal msg="Bootstrap failed to complete: waiting for Kubernetes API: context deadline exceeded"
```

* [Install container artifacts][2] contain both `.openshift_install.log` and the gathered log bundle.
* [Teardown][3] has [container logs][3.5], but no `.openshift_install.log`.  Presumably because this PR doesn't include the log shuffling #7936 adds on top.
* [Virtual IP teardown][4] has kind-of opaque `curl` output, but container logs are there.
* [DNS teardown][5] has a reasonable container log.

So I think we're in a pretty good space here.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3
[2]: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3/artifacts/e2e-vsphere/ipi-install-vsphere/
[3]: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3/artifacts/e2e-vsphere/ipi-deprovision-vsphere/
[3.5]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3/artifacts/e2e-vsphere/ipi-deprovision-vsphere/container-logs/ipi-deprovision-vsphere.log
[4]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3/artifacts/e2e-vsphere/ipi-deprovision-vsphere-vips/container-logs/ipi-deprovision-vsphere-vips.log
[5]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/7935/rehearse-7935-pull-ci-openshift-installer-master-e2e-vsphere/3/artifacts/e2e-vsphere/ipi-deprovision-vsphere-dns/container-logs/ipi-deprovision-vsphere-dns.log

**Comment by wking** [comment]
> /hold for #7931

/hold cancel

No need, since [this pivot][1].

[1]: https://github.com/openshift/release/pull/7935#issuecomment-604136798

**Pull request merged**